### PR TITLE
Added Field type reference

### PR DESCRIPTION
**Pull request by DavidLambauer**
Added all the common field types that can be used in the system.xml

## Purpose of this pull request

This pull request (PR) ...

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:270]:
```suggestion
Each field has the default type of ´text´ to display its frontend input type. The following input field types can be found and used in the core at this time :
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:275]:
```suggestion
| ´select´      | A common HTML select. Only one item can be selected.                                                              |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:276]:
```suggestion
| ´multiselect´ | A common HTML select. Multiple items can be selected at once.                                                     |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:279]:
```suggestion
| ´textarea´    | A common HTML textarea.                                                                                           |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:280]:
```suggestion
| ´button´      | A common HTML button. It is necessary to declare a custom ´frontend_model´ in order to use the button field type. | 
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:281]:
```suggestion
| ´image´       | Possible image upload. The nodes ´backend_model´, ´upload_dir´ and ´base_url´ need to be defined as well.        |
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
You need to update the description to the proper format. Please put the description in the "Pull request purpose", add full URL links to affected pages and remove the "Links to Magento Source Code"

As 2.3 documentation is identical but not symlinked, you will need to apply the corrected changes to the 2.3 file as well.

You also have some Lint errors (trailing spaces on line 279), these will need to be corrected. It is advised to run -
mdl --style=_checks/styles/style-rules-prod --ignore-front-matter --git-recurse -- .
before commiting.

**Comment by DavidLambauer** [comment]
Well, I made this whole PR via ´edit this site on github´ link. I won't be able to finish the changes you asked for in near time. Maybe you can fix it on your own?

**Comment by BarnyShergold** [comment]
Okay - I can update the description - but for the proposed changes, you just need to click the "Commit Suggestion" on this PR and Github will update. The rest I can do on Git if you have no way to correct.

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:275]: It should also be mentioned that this covers a YES/NO and it requires a source_model to manage the items on the list

EG - Normal dropdown, may need custom `source_model` - also used for Yes/No. See `Magento\Search\Model\Adminhtml\System\Config\Source\Engine` for an example

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:276]: Also requires a source_model

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:277]: Also mention that the value is encrypted and forcibly changing type using "Inspect Element" in browser will not show the value

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:278]: Incorrect definition : Allows a time to be set. This is done by using 3 dropdowns - Hour, minute and second.

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:279]: Would suggest that this be moved to under 'text'

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [reply src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:280]: Would also recommend mentioning : Requires custom front-end model to define the button text and the action. See `Magento\ScheduledImportExport\Block\Adminhtml\System\Config\Clean` for an example

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md
Comment (BarnyShergold) [reply src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:281]: 'password'
'file'
'label'
'allowspecific'

Need defining. Password is different to obscure.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

**Comment by BarnyShergold** [comment]
@DavidLambauer - I have added a lot of additional changes to be made and the 2.4 version of the file needs doing.

I accidentally created a second PR as this one wasn't linked to the issue (which is why I have made other suggestions). Therefore, if you want me to complete the work, I can arrange for THIS PR to be closed and use my PR to solve the issue.

The only downside is that you won't get any contributor points for this. Perhaps we can swap roles? If you were to review MY PR you would score points but as a Maintainer? (On DevDocs Maintainer = Contributor)

I'll leave this open for a few days

**Comment by DavidLambauer** [comment]
All good. I don't contribute for the reason of "points", so feel free to continue in the new PR.

**Comment by BarnyShergold** [comment]
> All good. I don't contribute for the reason of "points", so feel free to continue in the new PR.

Okay @DavidLambauer - that's great and thanks for the understanding . I'll get the PRs sorted and then ask you to review - that way it all gets done and your valuable input will be used :)

**Comment by meker12** [comment]
@BarnyShergold  I believe this can be closed because changes merged in PR #7948. Please confirm.

**Comment by BarnyShergold** [comment]
Closed because changes merged in PR #7948.

**Pull request closed without merging**
